tests/integration/helpers.py

```diff
@@ -267,6 +267,37 @@ async def wait_for(
     raise TimeoutError()
 
 
+async def generate_client_from_application(ops_test, jenkins_unit, address) -> UnitWebClient:
```

Comment:
You are actually generating it from the unit, not the application. Can you pass the application as an arg instead of the unit? Note that `jenkins_unit: Unit = application.units[0]` is always executed before invoking this method

Reply:
You can also get the address `ops.test.model` and the application name, so you can drop the `address` arg